dotnet/src/Connectors/Connectors.AssemblyAI/AudioToText/AssemblyAIAudioToTextService.cs

```diff
@@ -45,19 +44,19 @@ public sealed class AssemblyAIAudioToTextService : IAudioToTextService
     }
 
     /// <inheritdoc />
-    public async Task<TextContent> GetTextContentAsync(
+    public async Task<IReadOnlyList<TextContent>> GetTextContentsAsync(
         AudioContent content,
         PromptExecutionSettings? executionSettings = null,
         Kernel? kernel = null,
-        CancellationToken cancellationToken = default)
+        CancellationToken cancellationToken = default
+    )
     {
         Verify.NotNull(content);
 
         string uploadUrl;
         if (content.Data is not null)
```

Comment:
content.Data is { Empty: false }

Reply:
At this level we should check whether `Data` is null, if it is null, it should mean the user created the `AudioContent` with `Uri`.
I can check whether the `content.Data.IsEmpty` inside the first null check. What would the expected behavior be in case it is empty? Return an empty array of text content, or throw an exception?

Reply:
I think current solution is good if data is null or non-null but empty it checks uri, if uri is also null it throws exception